searchlib/src/vespa/searchlib/transactionlog/domainpart.cpp

```diff
@@ -527,121 +516,84 @@ DomainPart::visit(SerialNumRange &r, Packet &packet)
 bool
 DomainPart::visit(FastOS_FileInterface &file, SerialNumRange &r, Packet &packet)
 {
-    bool retval(true);
-    if ( ! file.IsOpened() ) {
-        retval = openAndFind(file, r.from() + 1);
+    if ( ! file.IsOpened() && ! openAndFind(file, r.from() + 1)) {
+        return false;
     }
-    if (retval) {
-        Packet newPacket;
-        Alloc buf;
-        for (bool full(false);!full && retval && (r.from() < r.to());) {
-            Packet::Entry e;
-            int64_t fPos = file.GetPosition();
-            retval = read(file, e, buf, false);
-            if (retval &&
-                e.valid() &&
-                (r.from() < e.serial()) &&
-                (e.serial() <= r.to())) {
-                try {
-                    full = addPacket(newPacket, e);
-                } catch (const std::exception & ex) {
-                    throw runtime_error(make_string("%s : Failed creating packet for visit %s(%" PRIu64 ") at pos(%" PRIu64 ", %" PRIu64 ")",
-                                                    ex.what(), file.GetFileName(), file.GetSize(), fPos, file.GetPosition()));
-                }
-                if ( !full ) {
-                    r.from(e.serial());
-                } else {
-                    if ( ! file.SetPosition(fPos) ) {
-                        throw runtime_error(make_string("Failed setting read position for file '%s' of size %" PRId64 " from %" PRId64 " to %" PRId64 ".",
-                                                        file.GetFileName(), file.GetSize(), file.GetPosition(), fPos));
-                    }
-                }
-            }
-        }
-        packet = newPacket;
+
+    packet = readPacket(file, r, TARGET_PACKET_SIZE, false);
+    if (!packet.empty()) {
+        r.from(packet.range().to());
     }
 
-    return retval;
+    return ! packet.empty();
 }
 
 void
-DomainPart::write(FastOS_FileInterface &file, const Packet::Entry &entry)
+DomainPart::write(FastOS_FileInterface &file, const IChunk & chunk)
 {
-    int64_t lastKnownGoodPos(byteSize());
-    int32_t crc(0);
-    uint32_t len(entry.serializedSize() + sizeof(crc));
     nbostream os;
-    os << static_cast<uint8_t>(_encoding.getRaw());
-    os << len;
-    size_t start(os.size());
-    entry.serialize(os);
-    size_t end(os.size());
-    crc = calcCrc(_encoding.getCrc(), os.data() + start, end - start);
-    os << crc;
-    size_t osSize = os.size();
-    assert(osSize == len + sizeof(len) + sizeof(uint8_t));
-
+    size_t begin = os.wp();
+    os << _encoding.getRaw();
```

Comment:
Consider adding comment about this being a placeholder value, to be patched up later.

Reply:
Fixed